### PR TITLE
Adding a reusable Editor component

### PR DESCRIPTION
**Pull request by gbowne1**
This commit adds boilerplate code for a reuseable editor component that can be used inside the app for micro-blogs, feed posts, for staff to enter in posts, etc. as a part of the Admin Dashboard.

This needs a resizeable text area to go along with it.  I did not have much time lately to complete this with the CSS so I reused some from the blank Dashboard component.

 Several things need to be done to Create a util or group of utilities for this component.  This needs a text area to go with it.

### CODE REVIEW

File: src/pages/Editor/Editor.jsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { Box, TextField, Button } from "@mui/material";
+import { FormControlLabel, FormGroup, Switch } from "@mui/material";
+import {
+  Save
```
Comment (pawel975) [inline src/pages/Editor/Editor.jsx:6]: Missing comma after 'save'.

**Comment by pawel975** [comment]
Does anyone has same Error?
image

File: src/pages/Editor/Editor.jsx (hunk 1 above)
Comment (gbowne1) [reply src/pages/Editor/Editor.jsx:6]: Oops. Missed it. 👍

**Comment by gbowne1** [comment]
Yeah, I get that too. I made an issue for that in case it doesnt.  Im not an expert in Prettier or ESLint.  Merging   Manuels PR should fix that.  Needs a review to merge.

**Comment by gbowne1** [comment]
I was looking for a good way to implement a text area, but that too might need its own custom imported component.  I am also assuming it needs some other text controls, etc. so 

Anyone wants to contribute one, just commit to this branch.

Several people I know are either using DraftJS or Lexical both have some problems with both of them.  Perhaps we make our own editor library?

**Review by manuel12 — APPROVED** [review]
I had the same issue as Pawel with the missing comma, now is good, we can enhance this editor later.

Btw my merge was about event tests, not related to eslint. I simply included changing the package version of eslint, which removed the error overlay in my local machine.

Is there any reason why we are not using the latest stable version of eslint?
It says in the website it's v8.39.0, as of now.

In any case, approving changes!

**Comment by gbowne1** [comment]
I think its a compatibility issue.  Yes theres a much newer parser and ESLint.  I use quick-lint-js but do not want to leave out people that use ESLint and Prettier. 

I only mentioned the update you made seeing that should help out.

I am by no mean expert on ESLint.  We will get it sorted though.

My component additions here have been kinda dry.  I havent had a lot of time but I figured I would help save time adding things we will use even if incomplete.  Anyone can commit to my branch and fix stuff, etc.

Lexical and DraftJS are a complete mess.  

Either way, I figured a editor we can use in several sections of the app would be helpful especially when it comes to the Groups feature.

Ideas @manuel12 @pawel975 ?

**Review by pawel975 — APPROVED** [review]

**Pull request merged**